### PR TITLE
HTCONDOR-838 import blahp

### PR DESCRIPTION
**Pull request by JaimeFrey**
Import blahp code-base into HTCondor repository.
https://opensciencegrid.atlassian.net/browse/HTCONDOR-838

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by timtheisen** [comment]
@matyasselmeci Since the condor-blahp RPM will replace the blahp RPM. Do we need a Conflcts: and/or Obsoletes:
Let us know.

**Comment by matyasselmeci** [comment]
Yes, if this is going into condor 9.5.0, you should do:
```
Provides: blahp = %{version}-%{release}
Obsoletes: blahp < 9.5.0
```
Conflicts aren't necessary.

File: build/packaging/srpm/condor.spec [hunk 1]
```diff
@@ -601,6 +593,21 @@ BOSCO provides an overlay system so the remote clusters appear to be a HTCondor
...
+Summary: BLAHP daemon
+Url: https://github.com/htcondor/BLAH
+Group: System/Libraries
+BuildRequires:  docbook-style-xsl, libxslt
+%if 0%{?rhel} >= 8 || 0%{?fedora}
+Requires: python3
+%else
+Requires: python >= 2.2
+%endif
+
```
Comment (timtheisen) [inline build/packaging/srpm/condor.spec:607]: Need to say that condor-blahp replaces blahp
```suggestion
Provides: blahp = %{version}-%{release}
Obsoletes: blahp < 9.5.0

```

**Review by timtheisen — CHANGES_REQUESTED** [review]
In the Debian control files. the htcondor package need the following additional declarations:
Provides: blahp
Conflicts: blahp
Replaces: blahp

You'll have to add them to the existing declarations for Conflicts and Replaces.

**Review by timtheisen — APPROVED** [review]
Looks good now.

**Pull request merged**